File: examples/python/chs_matching_example1.py
#!/usr/bin/env python
#encoding=utf-8
import sys
import os.path
module_dir = os.path.dirname(os.path.realpath(__file__))
sys.path.append(module_dir+'/../../lib/')
from tencent_ai_texsmart import *

print('Creating and initializing the NLU engine...')
engine = NluEngine(module_dir + '/../../data/nlu/kb/', 1)
if engine is None:
    sys.exit()

print(u'=== Text Matching ===')
str1 = u"我非常喜欢这只小狗";
str2 = u"我很爱这条狗";
output = engine.match_text(str1, str2)
if output is None or output.size() < 1:
    print(u'Error occurred in text matching')
    sys.exit()

print(u'text1: {0}'.format(str1))
print(u'text2: {0}'.format(str2))
print(u'Matching score: {0}'.format(output.score_at(0)))

File: examples/java/src/ChsMatchingExample1.java
import tencent.ai.texsmart.*;

public class ChsMatchingExample1 {
	public static void main(String[] args) {
		String userDir = System.getProperty("user.dir");
		String dataDir = args.length >= 1 ? args[0] : userDir + "/../../data/nlu/kb/";

		System.out.println("Creating and initializing the NLU engine...");
		NluEngine engine = new NluEngine();
		int workerCount = 4;
		boolean ret = engine.init(dataDir, workerCount);
		if(!ret) {
			System.out.println("Failed to initialize the engine");
			return;
		}

		System.out.println("=== Text Matching ===");
		String text1 = "我非常喜欢这只小狗";
		String text2 = "我很爱这条狗";
		TextMatchingOutput output = engine.matchText(text1, text2);
		if(output == null || output.size() < 1) {
			System.out.printf("Error occurred in text matching\n");
			return;
		}

		System.out.printf("Text-1: %s\nText-2: %s\nMatching score: %f\n",
				text1, text2, output.scoreAt(0));
	}
}
